src/Tribe/Views/V2/View.php

```diff
@@ -755,25 +755,37 @@ public function next_url( $canonical = false, array $passthru_vars = [] ) {
 
 		$url = $this->get_url();
 
+		$query_args = [];
+
 		if ( ! empty( $passthru_vars ) ) {
 			// Remove the pass-thru vars, we'll re-apply them to the URL later.
 			$url = remove_query_arg( array_keys( $passthru_vars ), $url );
 		}
 
 		// Make sure the view slug is always set to correctly match rewrites.
-		$url = add_query_arg( [ 'eventDisplay' => $this->slug ], $url );
+		$query_args['eventDisplay'] = $this->slug;
 
-		$url = $this->has_next_event ?
-			add_query_arg( [ $this->page_key => $this->url->get_current_page() + 1 ], $url )
-			: '';
+		if ( $this->has_next_event ) {
+			$query_args[ $this->page_key ] = $this->url->get_current_page() + 1;
+		} else {
+			$query_args = [];
+			$url        = '';
+		}
 
 		if ( ! empty( $url ) && $canonical ) {
 			$url = tribe( 'events.rewrite' )->get_clean_url( $url );
 		}
 
 		if ( ! empty( $passthru_vars ) && ! empty( $url ) ) {
 			// Re-apply the pass-thru query arguments.
-			$url = add_query_arg( $passthru_vars, $url );
+			$query_args = array_merge( $query_args, $passthru_vars );
+		}
+
+		$query_args = $this->filter_query_args( $query_args, $url );
+		$query_args = array_filter( $query_args );
+
+		if ( ! empty( $url ) || ! empty( $query_args ) ) {
+			$url = add_query_arg( $query_args, $url );
```

Comment:
:no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: f47045edc2a3259714bb50e8a4dbb210</sub>